File: local/src/main/java/com/concessions/local/network/dto/MenuDTO.java
package com.concessions.local.network.dto;

import java.io.Serializable;
import java.util.List;
import java.util.Objects;

/**
 * DTO for BaseMenu entity.
 * Includes a list of MenuItemDTOs for its menuItems relationship.
 */
public class MenuDTO implements Serializable {
    
    private Long id;
    private String name;
    private String description;
    private Long organizationId;
    
    // Relationship mapped to DTO
    private List<MenuItemDTO> menuItems;

    public MenuDTO() {
    }

    // --- Getters and Setters ---

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Long getOrganizationId() {
        return organizationId;
    }

    public void setOrganizationId(Long organizationId) {
        this.organizationId = organizationId;
    }

    public List<MenuItemDTO> getMenuItems() {
        return menuItems;
    }

    public void setMenuItems(List<MenuItemDTO> menuItems) {
        this.menuItems = menuItems;
    }

    // --- Utility Methods ---
    
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        MenuDTO menuDTO = (MenuDTO) o;
        return Objects.equals(id, menuDTO.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: app/lib/models/dto/menu_dto.dart
import 'menu_item_dto.dart';

class MenuDTO {
  final int id;
  final String name;
  final String? description;
  final int organizationId;
  final List<MenuItemDTO> menuItems;

  MenuDTO({
    required this.id,
    required this.name,
    this.description,
    required this.organizationId,
    required this.menuItems,
  });

  factory MenuDTO.fromJson(Map<String, dynamic> json) {
    return MenuDTO(
      id: json['id'],
      name: json['name'] ?? '',
      description: json['description'],
      organizationId: json['organizationId'],
      menuItems: (json['menuItems'] as List)
          .map((i) => MenuItemDTO.fromJson(i))
          .toList(),
    );
  }
}